module/zfs/vdev.c

```diff
@@ -2811,8 +2848,19 @@ vdev_stat_update(zio_t *zio, uint64_t psize)
 				vs->vs_self_healed += psize;
 		}
 
-		vs->vs_ops[type]++;
-		vs->vs_bytes[type] += psize;
+		/*
+		 * The bytes/ops/histograms are recorded at the leaf level and
+		 * aggregated into the higher level vdevs in vdev_get_stats().
+		 */
+		if (vd->vdev_ops->vdev_op_leaf) {
+			vs->vs_ops[type]++;
+			vs->vs_bytes[type] += psize;
+			vs->vs_queue_histo[zio->io_priority]
+						[HISTO(zio->io_delta)]++;
```

Comment:
I was going off the comment in the struct, which I (perhaps incorrectly) interpreted as queue time only:
```
struct zio {
    ...
    hrtime_t        io_delta;       /* vdev queue service delta */
```
However, looking further at the code it doesn't seem right.  I would expect the queue ending time to be calculated at the very beginning of `zio_vdev_io_start()`, since that would include the time spent in the active queue.  I will try using that calculation instead.